Normalize uri template name to be a valid regex groupname

Postremus opened the pull request:
See https://github.com/quarkusio/quarkus/issues/25258#issuecomment-1113714298 for a description of what is allowed in a uri template.

These changes add support for `_-.\d` in the component's name.

Previosly, the component name was directly used as a capture group name.
The above mentioned characters however are not allowed at some positions in groupnames however.

Therefore I added a simple normalization, making sure the component name only contains characters in the range `A-Za-z0-9`.

Also, I added support for spaces in before and after the name, and before and after the regex part.
The ABNF (see comment above) allows WSP in these cases, which is 0x09 (HTAB) or 0x20 (SPACE). I don't believe however that tabs are that often used inside an `@Path`.

fixes #25258

gsmet commented on independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/mapping/URITemplate.java:
```diff
@@ -124,38 +125,52 @@ public URITemplate(String template, boolean prefixMatch) {
+                regexAggregator.append("(?<").append(normalizedCaptureGroupName).append(">")
+                        .append(component.literalText.trim())
                         .append(")");
+                groupAggregator.add(normalizedCaptureGroupName + "");
```
Might be stupid as I haven't looked at the whole thing but could we simply use index based groups such as `group1` instead of generating something complex?

stuartwdouglas replied:
I think for valid names you can reference them later in the regex, so you definitely need to preserve the valid ones. For invalid ones I think anything unique would be fine.

gsmet replied:
But you use the pattern to reference them? It looks extremely weird.

stuartwdouglas replied:
So technically if you have @Path("someName:{someRegex}") on the class, you can then use @Path("otherName:{\\ksomeName}") to match whatever the original regex matched.

I don't think this is very common, but its why you need to actually use the name in the regex rather than just an index.

Postremus replied:
I implemented the requested changes. Now the groupname is by default the component name, and only changed to be `group1` (and so on) if it is invalid.

gsmet commented:
@Postremus is it ready for review? @stuartwdouglas any chance you could have a look, we build the 2.9.0.Final core artifacts tomorrow morning Paris time.

Postremus committed:
Normalize uri template name to be a valid regex groupname
these changes now also correctly permit space characters in the Path annotation, and matching to a pathparam without the spaces

stuartwdouglas reviewed (approved):
LGTM assuming the tests pass
